sources/packages/backend/apps/api/src/services/application/application.service.ts

```diff
@@ -1543,6 +1544,90 @@ export class ApplicationService extends RecordDataModelService<Application> {
     });
   }
 
+  /**
+   * Gets all eligible application that can be requested for application
+   * offering change.
+   * @param locationId location id.
+   * @param paginationOptions options to execute the pagination.
+   * @returns list of eligible application that can be requested for
+   * application offering change.
+   */
+  async getEligibleApplicationOfferingChangeRecords(
+    locationId: number,
+    paginationOptions: PaginationOptions,
+  ): Promise<PaginatedResults<Application>> {
+    const applicationQuery = this.repo
+      .createQueryBuilder("application")
+      .select([
+        "application.applicationNumber",
+        "application.id",
+        "currentAssessment.id",
+        "offering.studyStartDate",
+        "offering.studyEndDate",
+        "student.id",
+        "user.firstName",
+        "user.lastName",
+      ])
+      .leftJoin("application.currentAssessment", "currentAssessment")
+      .leftJoin("currentAssessment.offering", "offering")
+      .leftJoin(
+        "application.applicationOfferingChangeRequest",
+        "applicationOfferingChangeRequest",
+      )
+      .innerJoin("application.student", "student")
+      .innerJoin("student.user", "user")
+      .where(
+        new Brackets((qb) =>
+          qb
+            .where("applicationOfferingChangeRequest.id IS NULL")
+            .orWhere(
+              "applicationOfferingChangeRequest.applicationOfferingChangeRequestStatus NOT IN (:...status)",
+              {
+                status: [
+                  ApplicationOfferingChangeRequestStatus.InProgressWithSABC,
+                  ApplicationOfferingChangeRequestStatus.InProgressWithStudent,
+                ],
+              },
+            ),
+        ),
+      )
+      .andWhere("application.location.id = :locationId", { locationId })
+      .andWhere("application.applicationStatus = :applicationStatus", {
+        applicationStatus: ApplicationStatus.Completed,
+      })
+      .andWhere("application.isArchived = false");
+
+    if (paginationOptions.searchCriteria) {
```

Comment:
`if(paginationOptions.searchCriteria?.trim())` ?

Reply:
we have been using `trim()` inside the query search for all the pagination, that's the reason I was also following the same. let me know the way devs prefer

Reply:
I agree with @dheepak-aot. There is no need to create the query if the `searchCriteria` is a bunch of white space and line terminator characters.

Reply:
so, we will need a ticket to update all the pagination query